src/Tuura/Concept/STG/Translation.hs

```diff
@@ -22,20 +22,20 @@ translate circuit signs =
     case validate signs circuit of
         Valid -> do
             let initStrs = map (\s -> (show s, (getDefined $ initial circuit s))) signs
-                arcStrs = nubOrd (concatMap handleArcs (groupSortOn snd (arcs circuit)))
+                allArcs = arcLists (arcs circuit)
+                arcStrs = nubOrd (concatMap handleArcs (groupSortOn snd allArcs))
                 invStrs = map genInvStrs (invariant circuit)
                 inputSigns = filter ((==Input) . interface circuit) signs
                 outputSigns = filter ((==Output) . interface circuit) signs
                 internalSigns = filter ((==Internal) . interface circuit) signs
             genSTG inputSigns outputSigns internalSigns arcStrs initStrs invStrs
-        Invalid errs ->
-            "Error. \n" ++ addErrors errs
+        Invalid errs -> addErrors errs
 
 handleArcs :: Show a => [([Transition a], Transition a)] -> [String]
-handleArcs arcLists = addConsistencyTrans effect n ++ concatMap transition arcMap
+handleArcs xs = addConsistencyTrans effect n ++ concatMap transition arcMap
         where
-            effect = snd (head arcLists)
-            effectCauses = map fst arcLists
+            effect = snd (head xs)
```

Comment:
I'm just worried whenever I see partial functions like `head` or `tail` and can't immediately see that the list can't be empty.

It's easy to write `head` but may be very hard to track down the cause of a runtime error!